arborator/classes/read_data.py

```diff
@@ -37,7 +38,10 @@ def get_file_length(self,f):
         :param f: string path to file
         :return: int
         '''
-        return int(os.popen(f'wc -l {f}').read().split()[0])
+        result = os.popen(f'wc -l {f}').read().split()
+        if len(result) == 0:
+            return 0
```

Comment:
This is possibly a larger discussion, but I'm not a fan of spawning a subprocess to run `wc`. I think some people would recommend something like this, but I would instead suggest doing the equivalent in Python: opening the file and counting the number of lines. The `f` variable in `wc -l {f}` also allows for code injection under certain circumstances. Imagine `{f}` is: `file.txt; rm -rf /` or something else.

Reply:
This was done for speed reasons as wc is much faster than python https://stackoverflow.com/questions/54106136/time-difference-between-counting-lines-in-a-file-from-python-unix 

wc -l : 0.005s
python : 0.54s

Your security concerns make sense, I think this can be easily addressed as the specific number of lines isn't crucial. This is addressed in the id_fix branch commit https://github.com/phac-nml/arborator/pull/19/commits/22171b2e0b3e4b023661f900ed066bbbee26e2c3